src/sentry/users/api/serializers/user.py

```diff
@@ -0,0 +1,401 @@
+from sentry.api.serializers import Serializer, register
+from sentry.api.serializers.types import SerializedAvatarFields
+from sentry.app import env
+from sentry.auth.elevated_mode import has_elevated_mode
+from sentry.hybridcloud.services.organization_mapping import organization_mapping_service
+from sentry.models.authidentity import AuthIdentity
+from sentry.models.organization import OrganizationStatus
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.models.organizationmembermapping import OrganizationMemberMapping
+from sentry.organizations.services.organization import RpcOrganizationSummary
+from sentry.users.models.authenticator import Authenticator
+from sentry.users.models.user import User
+from sentry.users.models.user_avatar import UserAvatar
+from sentry.users.models.user_option import UserOption
+from sentry.users.models.useremail import UserEmail
+from sentry.users.models.userpermission import UserPermission
+from sentry.users.models.userrole import UserRoleUser
+from sentry.users.services.user import RpcUser
+from sentry.utils.avatar import get_gravatar_url
+
+T = TypeVar("T", bound=Model)
+
+
+def manytoone_to_dict(
+    queryset: QuerySet[T], key: str, filter_func: Callable[[Any], bool] | None = None
+) -> MutableMapping[Any, list[T]]:
+    result = defaultdict(list)
+    for row in queryset:
+        if filter_func and not filter_func(row):
+            continue
+        result[getattr(row, key)].append(row)
+    return result
+
+
+class _UserEmails(TypedDict):
+    id: str
+    email: str
+    is_verified: bool
+
+
+class _Organization(TypedDict):
+    slug: str
+    name: str
+
+
+class _Provider(TypedDict):
+    id: str
+    name: str
+
+
+class _Identity(TypedDict):
+    id: str
+    name: str
+    organization: _Organization
+    provider: _Provider
+    dateVerified: datetime
+    dateSynced: datetime
+
+
+class _UserOptions(TypedDict):
+    theme: str  # TODO: enum/literal for theme options
+    language: str
+    stacktraceOrder: int  # TODO enum/literal
+    defaultIssueEvent: str
+    timezone: str
+    clock24Hours: bool
+    prefersIssueDetailsStreamlinedUI: bool
+
+
+class UserSerializerResponseOptional(TypedDict, total=False):
+    # NOTE: There is a bug here where trying to move these fields to
+    # UserSerializerResponse and using NotRequired. "identities" is marked as
+    # required for places where UserSerializerResponse is used as a field (e.g
+    # OrganizationMemberResponse).
+    identities: list[_Identity]
+    avatar: SerializedAvatarFields
+    authenticators: list[Any]  # TODO: find out what type this is
+    canReset2fa: bool
+
+
+class UserSerializerResponse(UserSerializerResponseOptional):
+    id: str
+    name: str
+    username: str
+    email: str
+    avatarUrl: str
+    isActive: bool
+    hasPasswordAuth: bool
+    isManaged: bool
+    dateJoined: datetime
+    lastLogin: datetime | None
+    has2fa: bool
+    lastActive: datetime | None
```

Comment:
The model says that `lastLogin` and `lastActive` are nullable fields but the openAPI spec says these are required fields, not sure if there's a source of truth/who to follow for this response

Reply:
I definitely interpreted the OpenApi failing job wrong, as I didn't realize the OpenApi spec is generated from the model serializer which I guess didn't exist until now (?????)

Reply:
That job fails anytime there is a delta which isn't great. I took a look at the output and because we've added more accurate types schema generation has better metadata to use and the schema has changed. The properties were always nullable, but our schema didn't reflect that because of limited information.